test/object-store/sync/flx_sync.cpp

```diff
@@ -48,25 +223,18 @@ TEST_CASE("flx: connect to FLX-enabled app", "[sync][flx][app]") {
     auto app = sync_manager.app();
 
     auto creds = create_user_and_log_in(app);
-    std::shared_ptr<SyncUser> user;
-    app->log_in_with_credentials(creds, [&](std::shared_ptr<SyncUser> user_arg, util::Optional<app::AppError> error) {
-        REQUIRE_FALSE(error);
-        REQUIRE(user_arg);
-        user = std::move(user_arg);
-    });
-    REQUIRE(user);
+    auto user = app->current_user();
 
     SyncTestFile config(app, bson::Bson{}, schema);
+
     auto realm = Realm::get_shared_realm(config);
     CHECK(!wait_for_download(*realm));
     CHECK(!wait_for_upload(*realm));
 
-    CHECK(realm->sync_session()->has_flx_subscription_store());
-    CHECK(realm->get_latest_subscription_set().size() == 0);
-    CHECK(realm->get_active_subscription_set().size() == 0);
+    CHECK(!realm->sync_session()->has_flx_subscription_store());
 }
 
-TEST_CASE("flx: no subscription store created for PBS app", "[sync][flx][app]") {
+TEST_CASE("flx: connect to PBS as FLX returns an error", "[sync][flx][app]") {
```

Comment:
How about the opposite of this test: what happens if you try to open a populated FLX Realm later in PBS mode?

Reply:
Added!